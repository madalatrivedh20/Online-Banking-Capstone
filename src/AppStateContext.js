import { createContext, useState, useContext } from "react";

const AppStateContext = createContext();

export const AppStateProvider = ({ children }) => {
  const [isAuthenticated, setIsAuthenticated] = useState(false);
  const [issocialAuthenticated, setIssocialAuthenticated] = useState(false);
  const [user, setUser] = useState({});
  const [allUsers, setAllUsers] = useState([]);

  const value = {
    isAuthenticated,
    setIsAuthenticated,
    user,
    issocialAuthenticated,
    setIssocialAuthenticated,
    setUser,
    allUsers,
    setAllUsers
  };
  return <AppStateContext.Provider value={value}>{children}</AppStateContext.Provider>;
};

const useAppState = () => {
  const context = useContext(AppStateContext);

  if (context === undefined) {
    throw new Error("useAppState must be used within AppStateContext");
  }

  return context;
};

export default useAppState;
