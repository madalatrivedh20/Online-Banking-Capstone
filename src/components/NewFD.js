import React, { useState, useRef } from 'react';
import validator from 'validator';
import { toast } from 'react-toastify';
import { useNavigate } from 'react-router-dom';
import FormHeader from './FormHeader';
import { createFD } from '../service/api';
import useAppContext from '../AppStateContext';


// Component to create new FD and update the balance

const NewFD = () => {
  const navigate = useNavigate();
  const { user } = useAppContext();
  const toastId = useRef(null);

  const [data, setData] = useState({
    accNo: user.accno,
    FDProduct: "",
    FDPeriod: "",
    amount: ""
  });

  const [checkbox, setCheckbox] = useState(false);

  const [loading, setLoading] = useState(false);

  const onChangeHandler = (e) => setData({ ...data, [e.target.name]: e.target.value });


  // Function invoked on submission of the form with the details of the deposit

  const onSubmitHandler = async (e) => {
    e.preventDefault();
    if (!validator.isNumeric(data.accNo)) return toast.error("Invalid Account Number");
    if (!data.FDProduct) return toast.error("Please select FD Product");
    if (!data.FDPeriod) return toast.error("Please select FD Period");
    if (!validator.isNumeric(data.amount)) return toast.error("Invalid Amount");
    if (!checkbox) return toast.error("Please agree to terms and conditions");


    toastId.current = toast.loading("Creating Fixed Deposit...");
    const { type, render } = await createFD({ ...data, userId: user.id, accno: user.accno, acctype: user.acctype });

    toast.update(toastId.current, {
      type,
      render,
      isLoading: false,
      autoClose: 5000,
      closeOnClick: true,
      pauseOnHover: true,
      draggable: true,
    });

    if (type === "success") {
      return navigate("/balance");
    } else {
      setLoading(false);
    }
  };
  ;

  return (
    <div className="forms">
      <FormHeader title="Fixed Deposit" />
      <div>
        <form onSubmit={onSubmitHandler}>
          <div className="row">
            <label>From Account</label>
            <input required type="text" placeholder="Enter your account no" name="accNo"
              disabled
              onChange={onChangeHandler}
              value={data.accNo} />
          </div>

          <div className="row">
            <label>FD Type</label>
            <select value={data.FDProduct}
              placeholder="Select FD Type"
              onChange={onChangeHandler}
              name="FDProduct">
              <option value={""}>Select FD Type</option>
              <option value={"Standard"}>Standard</option>
              <option value={"Corporate"}>Corporate</option>
              <option value={"Cumulative"}>Cumulative</option>
              <option value={"Tax-Saving"}>Tax-Saving</option>
            </select>
          </div>

          <div className="row">
            <label>FD Period</label>
            <select value={data.FDPeriod}
              placeholder="Select FD Period"
              onChange={onChangeHandler}
              name="FDPeriod">
              <option value={""}>Select FD Period</option>
              <option value={"3 Months"}>3 Months</option>
              <option value={"6 Months"}>6 Months</option>
              <option value={"9 Months"}>9 Months</option>
              <option value={"12 Months"}>12 Months</option>
            </select>
          </div>

          <div className="row">
            <label>Amount</label>
            <input required
              type="text"
              name="amount"
              onChange={onChangeHandler}
              placeholder="Enter amounr"
              value={data.amount} />
          </div>

          <div className="row" style={{ flexDirection: "row" }}>
            <input required
              style={{ display: "inline", width: "min-content", marginLeft: '50px' }}
              value={checkbox}
              name="checkbox"
              id="checkbox"
              type={"checkbox"}
              onChange={e => setCheckbox(e.target.checked)}
              placeholder="Enter Remarks" />
            <label style={{ paddingLeft: '5px', color: "black" }}>Please agree to the terms and conditions</label>
          </div>

          <div id="button" className="row">
            <button>Create Fixed Deposit</button>

          </div>
        </form>

      </div >
    </div >
  );
};

export default NewFD;