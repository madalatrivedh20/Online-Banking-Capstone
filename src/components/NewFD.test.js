import { render, screen, waitForElementToBeRemoved } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import NewFD from './NewFD';

import {
  BrowserRouter as Router,
  Routes,
  Route
} from "react-router-dom";
import { AppStateProvider } from '../AppStateContext';

it('Snapshot Test', () => {
  const component = renderer.create(
    <AppStateProvider>
      <Router>
        <NewFD />
      </Router>
    </AppStateProvider>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders the page', async () => {
  render(
    <AppStateProvider>
      <Router>
        <NewFD />
      </Router>
    </AppStateProvider>
  );

  expect(screen.getByRole("heading")).toHaveTextContent(/Fixed Deposit/);
  expect(screen.getByPlaceholderText("Enter your account no")).toBeInTheDocument();
  expect(screen.getByPlaceholderText("Select FD Type")).toBeInTheDocument();
  expect(screen.getByPlaceholderText("Select FD Period")).toBeInTheDocument();
  expect(screen.getByPlaceholderText("Enter amounr")).toBeInTheDocument();
  expect(screen.getByRole("button", { name: "Create Fixed Deposit" })).toBeInTheDocument();
});