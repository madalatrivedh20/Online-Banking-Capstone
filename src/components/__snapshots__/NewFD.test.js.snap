// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test 1`] = `
<div
  className="forms"
>
  <div>
    <h2
      id="headerTitle"
    >
      Fixed Deposit
    </h2>
  </div>
  <div>
    <form
      onSubmit={[Function]}
    >
      <div
        className="row"
      >
        <label>
          From Account
        </label>
        <input
          disabled={true}
          name="accNo"
          onChange={[Function]}
          placeholder="Enter your account no"
          required={true}
          type="text"
        />
      </div>
      <div
        className="row"
      >
        <label>
          FD Type
        </label>
        <select
          name="FDProduct"
          onChange={[Function]}
          placeholder="Select FD Type"
          value=""
        >
          <option
            value=""
          >
            Select FD Type
          </option>
          <option
            value="Standard"
          >
            Standard
          </option>
          <option
            value="Corporate"
          >
            Corporate
          </option>
          <option
            value="Cumulative"
          >
            Cumulative
          </option>
          <option
            value="Tax-Saving"
          >
            Tax-Saving
          </option>
        </select>
      </div>
      <div
        className="row"
      >
        <label>
          FD Period
        </label>
        <select
          name="FDPeriod"
          onChange={[Function]}
          placeholder="Select FD Period"
          value=""
        >
          <option
            value=""
          >
            Select FD Period
          </option>
          <option
            value="3 Months"
          >
            3 Months
          </option>
          <option
            value="6 Months"
          >
            6 Months
          </option>
          <option
            value="9 Months"
          >
            9 Months
          </option>
          <option
            value="12 Months"
          >
            12 Months
          </option>
        </select>
      </div>
      <div
        className="row"
      >
        <label>
          Amount
        </label>
        <input
          name="amount"
          onChange={[Function]}
          placeholder="Enter amounr"
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="row"
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <input
          id="checkbox"
          name="checkbox"
          onChange={[Function]}
          placeholder="Enter Remarks"
          required={true}
          style={
            Object {
              "display": "inline",
              "marginLeft": "50px",
              "width": "min-content",
            }
          }
          type="checkbox"
          value={false}
        />
        <label
          style={
            Object {
              "color": "black",
              "paddingLeft": "5px",
            }
          }
        >
          Please agree to the terms and conditions
        </label>
      </div>
      <div
        className="row"
        id="button"
      >
        <button>
          Create Fixed Deposit
        </button>
      </div>
    </form>
  </div>
</div>
`;
